azurerm/resource_arm_key_vault_certificate.go

```diff
@@ -19,14 +19,70 @@ import (
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
 )
 
+//todo refactor and find a home for this wayward func
+func resourceArmKeyVaultChildResourceImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(*ArmClient).keyVaultClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseKeyVaultChildID(d.Id())
+	if err != nil {
+		return []*schema.ResourceData{d}, fmt.Errorf("Error Unable to parse ID (%s) for Key Vault Child import: %v", d.Id(), err)
+	}
+
+	list, err := client.ListComplete(ctx, utils.Int32(1))
+	if err != nil {
+		return []*schema.ResourceData{d}, fmt.Errorf("Error Unable to list Key Vaults for Key Vault Child import: %v", err)
+	}
+
+	for list.NotDone() {
+		v := list.Value()
+		if v.ID == nil {
+			log.Printf("[DEBUG] Key Vault Child import: v.ID was nil, continuing")
+			continue
+		}
+
+		vid, err := parseAzureResourceID(*v.ID)
+		if err != nil {
+			log.Printf("[DEBUG] Key Vault Child import: unable to parse v.ID (%s): %v", *v.ID, err)
+			continue
+		}
+		resourceGroup := vid.ResourceGroup
+		name := vid.Path["vaults"]
+
+		//resp does not appear to contain the vault properties, so lets fech them
+		get, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			log.Printf("[DEBUG] Key Vault Child import: Error making Read request on KeyVault %q (Resource Group %q): %+v", name, resourceGroup, err)
+			continue
+		}
+
+		if get.ID == nil || get.Properties == nil || get.Properties.VaultURI == nil {
+			log.Printf("[DEBUG] Key Vault Child import: KeyVault %q (Resource Group %q) has nil ID, properties or vault URI", name, resourceGroup)
+			continue
+		}
+
+		if id.KeyVaultBaseUrl == *get.Properties.VaultURI {
+			d.Set("vault_id", *get.ID)
```

Comment:
we shouldn't need to de-reference this here - d.Set should take care of that?

Reply:
```suggestion
			d.Set("vault_id", get.ID)
```